example-app/SantokuApp/src/bases/ui/modal/ModalBackdrop.tsx

```diff
@@ -0,0 +1,85 @@
+import {composePressableStyles} from 'bases/core/utils/composePressableStyles';
+import React, {useMemo} from 'react';
+import {Modal as RNModal, ModalProps, Pressable, PressableProps, StyleSheet, ViewProps} from 'react-native';
+import Reanimated, {WithTimingConfig} from 'react-native-reanimated';
+
+import {useModalBackdrop} from './useModalBackdrop';
+
+export const DEFAULT_BACKGROUND_COLOR = 'black';
+export const DEFAULT_OPACITY = 0.4;
+export const DEFAULT_FADE_IN_DURATION = 300;
+export const DEFAULT_FADE_OUT_DURATION = 150;
+
+// React Native ReanimatedのLayout Animationsを使用すると↓の不具合が発生するため、指定できないようにしています。
+// https://github.com/software-mansion/react-native-reanimated/issues/2906
+export type ModalBackdropProps = Omit<Reanimated.AnimateProps<ViewProps>, 'exiting' | 'entering'> & {
+  isVisible: boolean;
+  onPress?: () => unknown;
+  afterFadeIn?: (finished?: boolean) => unknown;
+  afterFadeOut?: (finished?: boolean) => unknown;
+  fadeInDuration?: number;
+  fadeOutDuration?: number;
+  opacity?: number;
+  pressableProps?: PressableProps;
+  modalProps?: ModalProps;
+  fadeInConfig?: WithTimingConfig;
+  fadeOutConfig?: WithTimingConfig;
+};
+
+export const ModalBackdrop: React.FC<ModalBackdropProps> = ({
```

Comment:
Personally, I feel a little that it would be better without it, since it is assumed to create and use parts that combine ModalXXX like BottomSheet.

However, there is nothing wrong with adding it, so I suppose it could be added, but I can't think of a good component name.

Any ideas?

Reply:
Sorry. I have no idea.